scripts/update_changelog_and_credits.py

```diff
@@ -131,10 +132,45 @@ def check_ordering_of_sections(release_summary_lines):
                     section.strip(), next_section.strip()))
 
 
-def update_changelog(release_summary_lines, current_release_version):
+def get_previous_version(version):
+    """Finds previous version given the current version.
+
+    Args:
+        version: str. The current version.
+
+    Returns:
+        str. The previous version.
+
+    Raises:
+        Exception. The version format is invalid or there is no previous
+            version for the given version.
+    """
+    if not bool(re.match(r'^\d+\.\d+\.\d+$', version)):
+        raise Exception('Invalid Version String: %s' % version)
+    first_version_num = int(version[:version.find('.')])
+    second_version_num = int(version[version.find('.') + 1: version.rfind('.')])
+    third_version_num = int(version[version.rfind('.') + 1:])
+
+    if third_version_num != 0:
+        return '%s.%s.%s' % (
+            first_version_num, second_version_num, third_version_num - 1)
+    else:
+        if second_version_num != 0:
+            return '%s.%s.%s' % (
+                first_version_num, second_version_num - 1, 9)
```

Comment:
This might not always work. We don't necessarily follow this pattern. If we have a larger release, we increment the second_version_num or third_version_num. Maybe a better solution is to get the tags and find the previous one.

Reply:
Also, what if the previous version was a hotfix branch?

Reply:
Really good point @nithusha21. I had missed this. I have updated the function to obtain previous version through tags and added a check to obtain correct version for hotfixes as well.